src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -16,15 +32,23 @@ export const TodoFilter = () => (
         type="text"
         className="input"
         placeholder="Search..."
+        value={query}
+        onChange={event => setQuery(event.target.value)}
       />
       <span className="icon is-left">
         <i className="fas fa-magnifying-glass" />
       </span>
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
+      {query.trim() && (
+        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+          <button
+            data-cy="clearSearchButton"
+            type="button"
+            className="delete"
+            onClick={() => setQuery('')}
+          />
+        </span>
+      )}
```

Comment:
The conditional rendering for the clear button is correct, but consider extracting the condition `query.trim()` into a variable for improved readability. This can make the JSX cleaner and the logic easier to understand.

Reply:
> The conditional rendering for the clear button is correct, but consider extracting the condition `query.trim()` into a variable for improved readability. This can make the JSX cleaner and the logic easier to understand.

Good point